app/src/app/store/packets/packets.ts

```diff
@@ -22,12 +24,19 @@ export const packetsSlice = createSlice({
             .addCase(actions.fetchPackets.rejected, (state, action) => {
                 state.fetchPacketsError = action.payload ?? null;
             })
-            .addCase(actions.fetchPacketById.fulfilled, (state, action) => {
+            .addCase(actions.fetchPacketMetadataById.fulfilled, (state, action) => {
                 state.packet = action.payload;
                 state.packetError = null;
             })
-            .addCase(actions.fetchPacketById.rejected, (state, action) => {
+            .addCase(actions.fetchPacketMetadataById.rejected, (state, action) => {
                 state.packetError = action.payload ?? null;
+            })
+            .addCase(actions.fetchFileByHash.fulfilled, (state, action) => {
+                state.fileUrl = action.payload;
```

Comment:
Isn't `fetchFileByHash` supposed to download the file content, rather than just provide the url`?